packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -218,6 +237,49 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
                 return;
               }
 
+              case 'tool-calls-generation': {
+                for (const toolCall of value.tool_calls) {
+                  controller.enqueue({
+                    type: 'tool-call',
+                    toolCallId: generateId(),
```

Comment:
The IDs need to be consistent for a single tool call. Does cohere provide an id? or can you use some other information (index) to identify the call?

Ideally only generate the id once for a particular tool call and then use the same id during streaming those tool call parts and the final tool call.

Reply:
good catch! 

cohere unfortunately does not provide an id ([source](https://docs.cohere.com/reference/chat#response.body.tool_calls)), the workaround is for the sdk to generate an id instead

they do provide an `index` during `tool-call-delta`, but they do not include it during the final `tool-call-generation` chunk, so I feel generating an `id` ourselves is better here since it allows developers to lookup a tool call in `chat_history` by a unique id